### DISCUSSION TITLE
USBで繋いでいないキーボードが動作しない

### DISCUSSION THREAD

**Discussion by degu-blank** [Yowkees/keyball#426] [Q&A]
お世話になります。
しばらく何の問題もなく使えていたのですが、
USBで繋いでいない方のキーボードが動作しなくなってしましました。

## 状態
- 右にUSBを繋いだらTRRSケーブルで繋いだ左が動作しない
- 左にUSBを繋いだらTRRSケーブルで繋いだ右が動作しない
- USBで繋いだ側のキーボードは動作する
- TRRSケーブルで繋いだ側のProMicroは光っていてロゴが表示されている
- 以前は問題なく動いていた(ProMicroへの書き込みは問題ないと思っている)
- #313
  - これを見て、左右のDATAの導通チェックをした
  - TRRSケーブルを繋いだ状態ならば、ピーと音がしたが、繋いでいない状態では音はしなかった
  - Dataの隣とショートしていないこと(音がならないこと)を確認した

お手数おかけしますが確認方法、修正方法がわからないのでご教示いただけますか。

<img width="570" alt="image" src="https://github.com/Yowkees/keyball/assets/136141560/3f0bf3e6-9b62-41c8-8880-1090a9147585">

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

症状から推測すると、TRSコネクタの半田付けが怪しいと思います。
TRSコネクタの半田が基板のパッドに馴染んで、山型になるように再半田してみてください。

以上、お手数ですがご対応お願いいたします。

**Reply by degu-blank** [reply]
@Yowkees 
回答ありがとうございます。

TRSコネクタの半田付けを山形になるように再半田してみましたが変わらずProMicroは光りますがキーが認識されない状態です。

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

そうなりますと、ProMicro裏かProMicro本体の半田付けで、接触が不十分な個所があると予想されます。
写真を見ると、一部はんだが少ない端子がありますので、半田を足してすべて同じようなサイズで山型になるように調整してください。
また写真には写っていませんが、ProMicro本体側もご確認お願いします。

以上、お手数ですがご対応お願いいたします。

**Reply by degu-blank** [reply]
ProMicro裏やProMicro本体のはんだ付けをやり直して、山型になるようにしましたが変わらずでした。
確認に必要な写真や導通チェックがありましたら教えていただけますか?

お手数おかけします。

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

左右の通信は、左右のProMicroのDATA線が繋がっているだけの単純な回路ですので、半田を再加熱しても直らないということはケーブルに問題があるかも知れません。TRSケーブルを別のものに変更して試すことはできますでしょうか？

お手数ですがご検討よろしくお願いいたします。

**Reply by degu-blank** [reply]
ProMicroのはんだ付けをもう一回やり直したところ治りました。
お手数おかけしました

**Comment by degu-blank** [comment]
@Yowkees 
何度も申し訳ありません。
 左右のキーボードの通信はできてタイプできるようになったのですが、
トラックボールの部分が反応しなくなってしまいました。（上記問題が発生する前までは問題なく動作していた）
以下の印の部分の導通チェックはしてみましたが、問題なく音はなりました。

また、ProMicro、トラックボールの部分の半田付けはやり直してみましたが、変わらない状況です。

![image](https://github.com/Yowkees/keyball/assets/136141560/e4237977-e746-4b96-ae8b-a44bf84c6468)
